### PR TITLE
Enable FxCop rules for Track 2 SDKs

### PR DESCRIPTION
**Pull request by pakrym**
I disabled some rules that I thought are inapplicable or hard to fix en masse.

@jsquire I didn't enable rules in EventHubs because of track 1 source that has a lot of failures.

### CODE REVIEW

File: sdk/core/Azure.Core/src/Pipeline/Policies/LoggingPolicy.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ namespace Azure.Core.Pipeline.Policies
 {
     public class LoggingPolicy : HttpPipelinePolicy
     {
-        private static readonly long s_delayWarningThreshold = 3000; // 3000ms
+        private const long s_delayWarningThreshold = 3000; // 3000ms
```
Comment (tg-msft) [inline sdk/core/Azure.Core/src/Pipeline/Policies/LoggingPolicy.cs]: Nit - if you have to update this PR, maybe change this from `s_delayWarningThreshold` to `DelayWarningThreshold`

File: sdk/core/Azure.Core/src/Http/HttpHeader.cs [hunk 2]
```diff
@@ -59,11 +61,15 @@ public static class Names
...
         public static class Common
+#pragma warning restore CA1034 // Nested types should not be visible
+#pragma warning restore CA1724 // Type name conflicts with standard namespace
         {
-            static readonly string s_applicationJson = "application/json";
-            static readonly string s_applicationOctetStream = "application/octet-stream";
-            static readonly string s_applicationFormUrlEncoded = "application/x-www-form-urlencoded";
+            const string s_applicationJson = "application/json";
+            const string s_applicationOctetStream = "application/octet-stream";
+            const string s_applicationFormUrlEncoded = "application/x-www-form-urlencoded";
```
Comment (tg-msft) [inline sdk/core/Azure.Core/src/Http/HttpHeader.cs]: Nit - if you have to update this PR, maybe change these to not use the `s_` prefix.

File: sdk/core/Azure.Core/src/Http/HttpHeader.cs [hunk 3]
```diff
@@ -12,7 +12,7 @@ public HttpHeader(string name, string value)
         {
             if (string.IsNullOrEmpty(name))
             {
-                throw new ArgumentException(nameof(name));
+                throw new ArgumentException("Name shouldn't be null or empty", nameof(name));
```
Comment (tg-msft) [inline sdk/core/Azure.Core/src/Http/HttpHeader.cs]: Make this `$"{nameof(name)} shouldn't be null or empty"` or at least change the case to match the parameter.

File: sdk/appconfiguration/Azure.ApplicationModel.Configuration/src/ConfigurationSettingParser.cs [hunk 4]
```diff
@@ -127,19 +128,19 @@ private static SettingBatch ParseSettingBatch(Response response, SettingSelector
             return new SettingBatch(settings, nextBatchUri);
         }
 
-        static readonly string s_link = "Link";
-        static readonly string s_after = "after=";
+        private const string s_link = "Link";
+        private const string s_after = "after=";
```
Comment (tg-msft) [inline sdk/appconfiguration/Azure.ApplicationModel.Configuration/src/ConfigurationSettingParser.cs]: Nit - naming with `s_` prefix again.

File: eng/Directory.Build.Data.props [hunk 5]
```diff
@@ -7,6 +7,26 @@
       SemVer 2.0 is supported by NuGet since 3.0.0 (July 2015) in some capacity, and fully since 3.5.0 (October 2016).
     -->
     <NoWarn>$(NoWarn);NU5105</NoWarn>
+    <!--
+      Disable some FxCop rules
+     -->
+    <NoWarn>
+      $(NoWarn);
```
Comment (tg-msft) [inline eng/Directory.Build.Data.props]: Should we file an issue to turn these back on someday when we have a chance to clean everything up?

**Review by tg-msft — APPROVED** [review]
Thank you for doing this!

File: eng/Directory.Build.Data.props (hunk 5 above)
Comment (pakrym) [reply eng/Directory.Build.Data.props]: Yes, I will do that

Comment (pakrym) [reply eng/Directory.Build.Data.props]: https://github.com/Azure/azure-sdk-for-net/issues/6763

**Review by KrzysztofCwalina — APPROVED** [review]

**Pull request merged**